packages/editor/src/core/components/Cell/InsertNew.tsx

```diff
@@ -31,6 +31,25 @@ const InsertNew: React.FC<{ parentCellId?: string }> = ({ parentCellId }) => {
     },
   });
 
+  React.useEffect(() => {
+    window?.addEventListener('keydown', handleKeyDown);
+    return () => {
+      window?.removeEventListener('keydown', handleKeyDown);
+    };
+  }, []);
```

Comment:
handleKeyDown is used before define, you should move the declaration up. 
You can declare it in `useEffect` directly. Make sure to add `handleInsertMode` to the deps array then.

Reply:
can you specify which file ? for better understanding

Reply:
i would move it somewhere in EditableEditor

Reply:
packages/editor/src/editor/EditableEditor.tsx -> this file doesn't have the redux connect to handle the useSetInsertMode hook . packages/editor/src/core/components/Editable/index.tsx -> shall we move the logic to this file?

Reply:
best wrap it in a component (`<GlobalHotkeys />`) and add that to the `<EditableEditor />` inside the `<Provider />`

then this `GlobalHotkeys` just returns null, but has a `useEffect` hook.

Reply:
@macrozone updated !